src/DurableTask.SqlServer.AzureFunctions/SqlDurabilityProvider.cs

```diff
@@ -218,13 +214,12 @@ public override bool TryGetTargetScaler(
             string connectionName,
             out ITargetScaler targetScaler)
         {
-            if (this.targetScaler == null)
+            if (this.singletonSqlMetricsProvider == null)
             {
-                var sqlMetricsProvider = new SqlMetricsProvider(this.service);
-                this.targetScaler = new SqlTargetScaler(hubName, sqlMetricsProvider);
+                this.singletonSqlMetricsProvider = new SqlMetricsProvider(this.service);
             }
 
-            targetScaler = this.targetScaler;
+            targetScaler = new SqlTargetScaler(hubName, this.singletonSqlMetricsProvider);
```

Comment:
Same comment about using the function ID to ensure uniqueness in the target scaler.